### ISSUE TITLE
don't prompt for port if there is only one

### ISSUE THREAD

**Issue by davidelang** [yann25/sonoff_autoflash#1]
As a suggestion, don't prompt for port if there is only one

This would probably involve making the ports array global

**Comment by ThomDietrich** [comment]
you could also use esptool to identify a valid ESP on ports available, ruling out the extra ones.

Instead of making the ports array global, it would be better to return it from the function.

**Comment by yann25** [comment]
I did not find an option to do this in the esptool documentation. Could you give me a hint? :) 

If esptool doesn't work I will implement davidlang's suggestion.

**Comment by ThomDietrich** [comment]
I'm not sure either 😃 You could do some testing. I suspect either the [chip id or the mac address](https://github.com/espressif/esptool/blob/master/esptool.py#L825-L844) to be definite (i got the idea from [here](https://github.com/espressif/esptool/pull/131/files).

One approach would be to check if the retrieved MAC is valid, otherwise rule out this port. This can be combined with @davidelang's idea 👍

**Comment by ThomDietrich** [comment]
Could also be interesting: https://github.com/espressif/esptool/blob/master/esptool.py#L175

**Comment by davidelang** [comment]
As a general rule, I really dislike the idea of sending anything to a serial 
port that you have not identified. you have no idea what is connected there, and 
what your probing is going to do to it.

Having it be optional is ok, but the default if there is more than one serial 
port available should be to ask which one to use

**Comment by ThomDietrich** [comment]
Fair point. However the functions are only reading registers, that should be uncritical... wdyt?

**Comment by davidelang** [comment]
If you are reading registers from the serial adapter it could be ok. But if you 
are trying to read registers from the ESP-8266, that can only happen if you 
first send some characters to the ESP to tell it what registers you want to read

**Comment by ThomDietrich** [comment]
Now that you say it, yes that's a problem. It was just an idea but okay, let's leave it aside for now. Good talk ;)

**Comment by davidelang** [comment]
It's good to have an autodetect option, but that shouldn't be the default.

for many people, it will work (we don't have that many things connected by 
serial ports any more, especially on laptops), but we don't want to make it the 
default and mess up people who do have other things connected.

**Comment by yann25** [comment]
I implemented the function to not promt if only one port is aviable. 

I think a good way would be to detect the aviable ports and give each port an index number, so that the user input is an int and not a string. This way there is less room for mistakes and less work for the user.

**Comment by davidelang** [comment]
I agree
